src/qz/installer/provision/ProvisionInstaller.java

```diff
@@ -129,6 +136,31 @@ private boolean invokeStep(Step step) throws Exception {
         return invoker.invoke();
     }
 
+    /**
+     * Loops through steps searching for a property which sets a custom websocket ports
+     */
+    public void setCustomPorts(Installer installer) {
+        for(Step step : steps) {
+            if(step.getType() == Type.PROPERTY) {
+               for(AbstractMap.SimpleEntry<String,String> pair : PropertyInvoker.parsePropertyPairs(step)) {
```

Comment:
Map.Entry may be easier to read here

Reply:
Better yet, instead of doing an ArrayList of entries, a Hashmap<String, String> would probably be better. Then for this part of the code, you do a `map.get(ArgValue.WEBSOCKET_SECURE_PORTS.getMatch())` instead of the for loop.

Reply:
Agreed, but we should warn if duplicates are found.  Working on it...

Reply:
Done via 37b2838702c4bfd73a1f3e90e16ec5bf16794885